src/mappings/staking.ts

```diff
@@ -139,6 +146,7 @@ export async function handleAddDelegation(
   await delegation.save();
   await updateIndexerCapacity(indexer, event);
   await updateMaxUnstakeAmount(indexer, event);
+  await updateStakeSummary(event, eraManager);
```

Comment:
make sure currentEra is only queried once in each handler, and can even provide cross handler cache for it via utilising global variable

Reply:
should I refactor the 5 functions above to meet this requirement? it's been 5 same queries before this query...

Reply:
Yes please